### ISSUE TITLE
Make model factory match Eloquent collection

### ISSUE THREAD

**Issue by jasonmccreary** [laravel/ideas#1914]
This _might_ be a bug, but I'm submitting it as an idea since there are a few possible solutions. Once I receive some directional feedback from the core team, I am glad to open a PR for the change.

Ideally, when multiple models are _created_ via a factory, it should match the Eloquent collection of the same models.

```php
public function testExample()
{
    $users = factory(\App\User::class, 2)->create();

    $this->assertEquals($users, User::all());
}
```

However, this assertion fails with the following (Laravel 6 + SQLite in-memory DB):

```diff
1) Tests\Feature\ExampleTest::testExample
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
             'fillable' => Array ()
             'hidden' => Array ()
             'connection' => 'sqlite'
-            'table' => null
+            'table' => 'products'
             'primaryKey' => 'id'
             'keyType' => 'int'
             'incrementing' => true
@@ @@
             'withCount' => Array ()
             'perPage' => 15
             'exists' => true
-            'wasRecentlyCreated' => true
+            'wasRecentlyCreated' => false
             'attributes' => Array (...)
             'original' => Array (...)
             'changes' => Array ()
@@ @@
             'fillable' => Array ()
             'hidden' => Array ()
             'connection' => 'sqlite'
-            'table' => null
+            'table' => 'products'
             'primaryKey' => 'id'
             'keyType' => 'int'
             'incrementing' => true
@@ @@
             'withCount' => Array ()
             'perPage' => 15
             'exists' => true
-            'wasRecentlyCreated' => true
+            'wasRecentlyCreated' => false
             'attributes' => Array (...)
             'original' => Array (...)
             'changes' => Array ()
```

Upon closer inspection of the code, it seems when a model factory is _created_ it does update the table connection and calls save in the [`FactoryBuilder::store` method](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Database/Eloquent/FactoryBuilder.php#L203).

Maybe this method could return a new, updated collection or use references to ensure the model is updated inline. I tinkered around with it a bit, but didn't get too far.

In the end, there are ways around this. But they require more test setup, so it would be nice if this _just worked_ as underlying methods like `assertViewHas` use this same equality assertion and as such result in a failure.

**Note:** these core assertion methods do work for single model factories created because underneath they use a `Model::is` check instead. My initial inclination was to add the same check for Eloquent collections. But doing so cures the symptom and not the underlying cause. In addition, it would be a breaking change as the strictness of the check is lessened.

**Comment by mfn** [comment]
I see the discrepancy but also wondering what's your use case? The example using assertion with `User::all()` is probably not the real story ;)

**Comment by jasonmccreary** [comment]
@mfn, the example serves as a quick recreation.

Nonetheless, `User::all()` is an example of code used within an app, so it follows that anytime you'd compare a collection of _created_ model factories against its corresponding eloquent collection of models you would receive this _false_ failure.

Basically, I can't readily compare the created models against _any query_ which resulted in the same set of models.

**Comment by codytooker** [comment]
@jasonmccreary have you found a solution to this?  Running into this now as well.

**Comment by jasonmccreary** [comment]
No. I usually end up "rebuilding" the query in the test or testing smaller portions. It don't run into this too often, but the example does still prove it happens.

**Comment by codytooker** [comment]
@jasonmccreary thanks for the response.  I guess I'll go that route as well.
